static/js/components/LimitSelector.tsx

```diff
@@ -0,0 +1,77 @@
+import React, { ChangeEvent, Dispatch, FC, SetStateAction } from "react";
+import { Col, Input, RadioInput, Row } from "@canonical/react-components";
+
+interface Props {
+  cpuLimit: string;
+  setCpuLimit: Dispatch<SetStateAction<string>>;
+  memoryLimit: string;
+  setMemoryLimit: Dispatch<SetStateAction<string>>;
+  measure: string;
+  setMeasure: Dispatch<SetStateAction<string>>;
+}
+
+const MIB = "MiB";
+const GIB = "GiB";
+
+const LimitSelector: FC<Props> = ({
+  cpuLimit,
+  setCpuLimit,
+  memoryLimit,
+  setMemoryLimit,
+  measure,
+  setMeasure,
+}) => {
+  const updateCpuLimit = (v: ChangeEvent<HTMLInputElement>) => {
+    setCpuLimit(v.target.value);
+  };
+
+  const updateMemoryLimit = (v: ChangeEvent<HTMLInputElement>) => {
+    setMemoryLimit(v.target.value);
+  };
+
+  return (
+    <>
+      <Input
+        id="cpuLimit"
+        name="cpuLimit"
+        type="number"
+        label="CPU limit"
+        onChange={updateCpuLimit}
+        value={cpuLimit}
+        stacked
```

Comment:
Can we add a help text for this ... Is it a value between 1 and 100 or 0 and 1?

Reply:
Looking into this, I just found out that this field should be able to accommodate either:

- a number, or
- a range of numbers, or
- a set of comma-separated integers

So a simple numeric input is not good, see: https://linuxcontainers.org/lxd/docs/master/reference/instance_options/#cpu-limits

I will use a `text` input and validate its content with regular expressions.

Reply:
Maybe three radio buttons to select the type of limit and a changing input based on which option was selected is better here?

Reply:
> a set of comma-separated integers
This could be checkboxes for each core that you can enable/disable

Reply:
Ok, the limits should be fixed too @ [888ddae](https://github.com/canonical/lxd-ui/commit/888ddaeaca03ce4bf85a1d7ea782c0b4281e4840)